internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -801,7 +803,9 @@ func (r *Reconciler) reconcileReferencedTemplate(ctx context.Context, in reconci
 // createErrorWithoutObjectName removes the name of the object from the error message. As each new Create call involves an
 // object with a unique generated name each error appears to be a different error. As the errors are being surfaced in a condition
 // on the Cluster, the name is removed here to prevent each creation error from triggering a new reconciliation.
-func createErrorWithoutObjectName(err error, obj client.Object) error {
+func createErrorWithoutObjectName(ctx context.Context, err error, obj client.Object) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Failed to create object", "object", klog.KObj(obj), "error", err.Error())
```

Comment:
I think the simplest solution is just a generic log every time this function is called, but I'm open to opinions on when this might not be desired.

Reply:
I think thats fair, except it not being obvious to log the error due to the function name 🤔 

But `createAndLogErrorWithoutObjectName` is also weird.

Reply:
Can we fix this by using log.WithCallDepth(1)?

Reply:
What about the following?
```go
log.WithCallDepth(1).Error(err, "Failed to create object", obj.GetObjectKind().GroupVersionKind().Kind, klog.KObj(obj))
```
* uses the Kind instead of just object (I think it should be always set, we set it explicitly in MD and all the unstructured stuff needs it). Would be good to double-check though.
* WithCallDepth(1) should work, but please verify
* I would use .Error instead of adding a error k/v pair

Reply:
The Call Depth is already set here as this case is using the topology logger (which sets its own call level). Right now it's logging the line number where the function call occurs which is the desired behaviour.

I've updated to use Kind (which I'm  mixed on because object makes it clearer to me which object the creation error refers to in this line) and changed to .Error() in the latest version.